### PR TITLE
MWPW-108925 [Blog Support] Article Ratings

### PR DESCRIPTION
**Pull request by prathyushanallapati**
Implement Review component in Milo.

Resolves: [MWPW-108925](https://jira.corp.adobe.com/browse/MWPW-108925)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/
- After: https://main--milo--prathyushanallapati.hlx.page/drafts/boyapati/review

### CODE REVIEW

**Comment by prathyushanallapati** [comment]
CC: @auniverseaway @chrischrischris

**Comment by prathyushanallapati** [comment]
@auniverseaway @chrischrischris @amitbikram

The code has been migrated from acom review to Milo. Please provide your reviews.

We are working on the UT and will create a new PR.

Thanks!!

**Review by fullcolorcoder — CHANGES_REQUESTED** [review]
Missing test coverage. There are some examples here https://github.com/adobecom/milo/tree/main/test/blocks

File: libs/blocks/list.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const blocks = [
   'how-to',
   'modal',
   'marquee',
+  'review',
```
Comment (chrischrischris) [inline libs/blocks/list.js:11]: Changes to list.js need to go into their own separate PR.  This is to ensure that enabling a milo block doesn't adversely affect consumers.

File: libs/blocks/review/review.css [hunk 2]
```diff
@@ -0,0 +1,274 @@
+:root {
+	--ratingBgImage: url('./img/star-outline.svg');
```
Comment (chrischrischris) [inline libs/blocks/review/review.css:2]: Please use spaces (default indent is 2)

**Review by chrischrischris — COMMENTED** [review]
Is there any code that was modified during the port (other than preacting it)?

Tests will also be required.

**Review by amauch-adobe — CHANGES_REQUESTED** [review]
I noticed a couple of things when I tried out the rating tool on all the different browsers (Safari, Chrome, Edge, Firefox).  The rating block will not show the feedback textbox unless you select 'Good, Poor, Very Poor' options, which I believe is the intended behavior, with that being said if you select one of those three options first and then quickly change your rating to 'Very Good or Excellent' option, the feedback text box still shows instead of hiding and then expects feedback to submit the rating.  I believe the feedback text box should go away after switching from the three lower ratings to the two higher star ratings.

A second concern was with the WCAG AA level contrast issue with the feedback box.  If the feedback box is on a plain white background like the testing URL you provided it will be too low of a contrast.  Granted this will be based on authoring, but something to take into account.

File: libs/blocks/review/review.css (hunk 2 above)
Comment (prathyushanallapati) [reply libs/blocks/review/review.css:2]: Sure, Thanks!!

File: libs/blocks/list.js (hunk 1 above)
Comment (prathyushanallapati) [reply libs/blocks/list.js:11]: will make the changes accordingly. Thanks!!

**Comment by prathyushanallapati** [comment]
> Is there any code that was modified during the port (other than preacting it)?
> 
> Tests will also be required.

@chrischrischris No code modifications has been done apart from preacting

**Comment by prathyushanallapati** [comment]
> I noticed a couple of things when I tried out the rating tool on all the different browsers (Safari, Chrome, Edge, Firefox). The rating block will not show the feedback textbox unless you select 'Good, Poor, Very Poor' options, which I believe is the intended behavior, with that being said if you select one of those three options first and then quickly change your rating to 'Very Good or Excellent' option, the feedback text box still shows instead of hiding and then expects feedback to submit the rating. I believe the feedback text box should go away after switching from the three lower ratings to the two higher star ratings.
> 
> A second concern was with the WCAG AA level contrast issue with the feedback box. If the feedback box is on a plain white background like the testing URL you provided it will be too low of a contrast. Granted this will be based on authoring, but something to take into account.

Thanks for the feedback.

The component has been built taking acom review component as the reference. Sample acom review component on dexter:
https://dexter-author.qa01.corp.adobe.com/content/dexter/us/en/sharath/review.html?wcmmode=disabled

The same functionality and design have been taken forward. Please let me know if these changes are to be made in this PR I can take forward.

Thanks!!

File: libs/blocks/review/review.css (hunk 2 above)
Comment (prathyushanallapati) [reply libs/blocks/review/review.css]: Done!!

File: libs/blocks/list.js (hunk 1 above)
Comment (prathyushanallapati) [reply libs/blocks/list.js]: Done!!

**Comment by chrischrischris** [comment]
@amauch-adobe The behavior with the comment block not disappearing is intentional.  It was in case a user had written something in there - though I don't remember the exact rationale for it.

**Review by chrischrischris — APPROVED** [review]
I think we will want to consider consolidating a bunch of these files, but I think we should land the PR as is and do that in a new one after this has been merged.

**Comment by amauch-adobe** [comment]
@prathyushanallapati, I realized the contrast issue isn't there for the rating feedback textbox at a second glance.  The Axe test struggled through it since it couldn't figure out the background color due to an overlapping issue.  So it was a false failure.  The contrast is correct for WCAG accessibility.  Sorry for the confusion.  Everything is good for me on my end.  

Although as the others have mentioned there should be mocha tests associated with the review block for code coverage.  All our PRs for new blocks require it.  If you could add those tests that would be great.  Otherwise, this is approved.

**Review by amauch-adobe — APPROVED** [review]

**Comment by nkthakur48** [comment]
@fullcolorcoder I have added few tests and we will continue adding more as part of new PRs. As mentioned by @chrischrischris there may be some refactoring once this PR is merged, and we will take care of adding updated UTs at that time as well

**Review by amauch-adobe — APPROVED** [review]
Good start for tests.  Approved for the added tests.

**Review by fullcolorcoder — APPROVED** [review]

**Comment by fullcolorcoder** [comment]
@[nkthakur48](https://github.com/nkthakur48) Sounds good, thanks!

**Comment by nkthakur48** [comment]
@auniverseaway @fullcolorcoder @chrischrischris @amauch-adobe Could you please merge this PR? Thank you so much!

**Review by amauch-adobe — APPROVED** [review]

**Comment by amauch-adobe** [comment]
@nkthakur48, I have merged the PR after everyone's reviews.  For future PRs in Milo that need to be associated as a release, you need to add the label "release".  Without the label, our system doesn't send out an email with the release notes from the PR to various stakeholders subscribed to the Milo emails.

**Comment by amauch-adobe** [comment]
@nkthakur48 so please reach out to the stakeholders to the ticket and make sure they know that the code has been merged into Milo.

**Comment by nkthakur48** [comment]
Sure thing @amauch-adobe. Thank you so much! 🙏
